torch/jit/_static_analysis.py

```diff
@@ -0,0 +1,43 @@
+"""
+Tools for static analysis of PyTorch JIT
+
+This is not intended to be imported directly; please use the exposed
+functionalities in `torch.jit`.
+"""
+
+from typing import Any, List
+
+import torch
+from torch import TensorType
+from torch._C import Graph
+
+
+def apply_input_props_using_example(graph: Graph, example_input: List[Any]):
```

Comment:
This helper only sets dtype and sizes. There are other properties such as strides, device, undefined or not, requires_grad or not. I would rename this or split into two smaller helpers `apply_sizes`, `apply_dtypes`.

Reply:
How do I get a `TensorType` object from a tensor in Python?